src/block-management/block-manager.js

```diff
@@ -164,14 +164,15 @@ export default class BlockManager extends React.Component<PropsType, StateType>
 	}
 
 	onChange( clientId: string, attributes: mixed ) {
-		// Update datasource UI
+		// Update Redux store
+		this.props.onChange( clientId, attributes );
+
+		// Change the data source
 		const index = this.getDataSourceIndexFromClientId( clientId );
 		const dataSource = this.state.dataSource;
-		const block = dataSource.get( index );
+		const block = { ...( dataSource.get( index ) ) };
```

Comment:
Ah sorry, I was trying to do another change here to see if I could just do a copy of the block and avoid conflicts with the reducer. Let me change that back.